File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <bits/stdc++.h>
using namespace std;
int ret;
void dfs(vector<int> numbers,int target, int index, int total){
    if(index==numbers.size()){
        if(total==target){
            ++ret;
        }
        return;
    }
    int sum1= total+numbers[index];
    int sum2= total-numbers[index];
    index++;
    dfs(numbers,target,index,sum1);
    dfs(numbers,target,index,sum2);
    return;
}
int solution(vector<int> numbers, int target) {
    dfs(numbers,target,0,0);
    return ret;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;
class Solution {
    int answer;
    public void dfs(int sum, int idx, int target, int[] numbers){
        if(idx == numbers.length){
            if(sum == target){
                ++answer;
            }
            return;
        }
        dfs(sum+numbers[idx],idx+1,target,numbers);
        dfs(sum-numbers[idx],idx+1,target,numbers);
    }
    public int solution(int[] numbers, int target) {
        dfs(0,0,target,numbers);
        return answer;
    }
}